src/TipToiYaml.hs

```diff
@@ -164,11 +170,37 @@ scriptCodes codes codeMap
             Nothing -> Left s 
             Just n -> Right (n::Word16)
 
+-- The following logic (for objectCodes) tries to use different object codes 
+-- for different projects, as far as possible. This makes the detection of not 
+-- having activated a book/product more robust.
+
+-- We could theoretically set: 
+--    objectCodeOffsetMax = lastObjectCode - firstObjectCode.
+-- This would assign perfectly usable object codes, and would minimize the 
+-- probability of object code collisions between products, but sometimes 
+-- object codes would wrap around from 16383 to 1000 even for small projects 
+-- which may be undesirable. We arbitrarily do not use the last 199 possible 
+-- offsets to avoid a wrap around in object codes for projects with <= 200 
+-- object codes. This does not impose any limit on the number of object codes 
+-- per project. Every project can always use all 15384 object codes.
+    objectCodeOffsetMax = lastObjectCode - firstObjectCode - 199
```

Comment:
Shouldn’t this be much larger? A wrap-around means that the `gme` file becomes very large (one word per oid code _in range_). Have you made sure that this is not a problem for the pen?